### PR TITLE
`jellyfish-wallet-encrypted`

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Passphrase protected storage, to securely store sensitive data like private key
Dependency for #133 

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/jellyfish-secured-storage/src/scrypt-storage.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+export class ScryptStorage {
+  readonly scryptProvider: ScryptProvider
+  readonly storage: Storage
+
+  constructor (scryptProvider: ScryptProvider, storage: Storage) {
+    this.scryptProvider = scryptProvider
+    this.storage = storage
+  }
+
+  async encrypt (data: Buffer, passphrase: string): Promise<void> {
```
Comment (ivan-zynesis) [inline packages/jellyfish-secured-storage/src/scrypt-storage.ts:72]: https://github.com/bitcoin/bips/blob/master/bip-0038.mediawiki#Encryption_when_EC_multiply_flag_is_not_used

File: package.json [hunk 2]
```diff
@@ -54,5 +55,8 @@
       "path": "packages/jellyfish/dist/jellyfish.umd.js",
       "limit": "1000 ms"
     }
-  ]
+  ],
+  "dependencies": {
+    "scryptsy": "^2.1.0"
+  }
```
Comment (fuxingloh) [inline package.json:61]: don't edit root package.json

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -14,6 +14,7 @@
 /packages/jellyfish-crypto/                     @fuxingloh
 /packages/jellyfish-json/                       @fuxingloh @canonbrother
 /packages/jellyfish-network/                    @fuxingloh
+/packages/jellyfish-secured-storage             @ivan-zynesis
```
Comment (fuxingloh) [inline .github/CODEOWNERS:17]:
```suggestion
/packages/jellyfish-wallet-encrypted            @ivan-zynesis
```

File: packages/jellyfish-secured-storage/__tests__/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+let inMemory: string | undefined
+const storage: Storage = {
+  getter: async () => inMemory,
+  setter: async (encrypted: string | undefined) => { inMemory = encrypted }
+}
+
+const scryptProvider = new SimpleScryptsy()
+
+// 32 bytes
+const samplePublicKey = 'e9873d79c6d87dc0fb6a5778633389f4e93213303da61f20bd67fc233aa33262'
```
Comment (fuxingloh) [inline packages/jellyfish-secured-storage/__tests__/index.test.ts:13]: Do you mean privateKey?

**Review by fuxingloh — COMMENTED** [review]
The direction looks good, it's agnostic enough for implementation on various platform. This review will take a while, I need to study the specification.

I think for the test you can mock how RN or other apps will implement it with a fake wallet in mind. Probably can use import mnemonic as dev dependencies.

Do document this heavily, it will help with the understanding when reviewing. Thanks!

File: packages/jellyfish-secured-storage/__tests__/index.test.ts (hunk 4 above)
Comment (ivan-zynesis) [reply packages/jellyfish-secured-storage/__tests__/index.test.ts:13]: Yes

Comment (ivan-zynesis) [reply packages/jellyfish-secured-storage/__tests__/index.test.ts:13]: Yes

**Comment by ivan-zynesis** [comment]
re-open to trigger oss bot

**Pull request closed without merging**